### DISCUSSION TITLE
Documentation Website

### DISCUSSION THREAD

**Discussion by morellodev** [TanStack/query#500] [General] [answered]
Hi @tannerlinsley, first of all thank you for your precious work!

I am opening this issue because I find your documentation very well written, but it is stored into a single README file (which gets longer and longer release after release). It becomes even more painful to read when looking for changes between different releases. Have you considered opening a documentation website using tools like Docusaurus – maybe supported by the community? I think it would be super useful.

Have a nice day!

**Comment by tannerlinsley** [comment, marked as answer]
I've considered it, but it's not a high priority yet. Thanks for the feedback! There will be one soon enough.

**Comment by joserocha3** [comment]
Hey @tannerlinsley, do you want some help with this? It would look much more professionalized with an official docs site. I'll have some time to start this within the next few weeks to help transition things if you're game.

Btw, love `react-query`, miles ahead of anything like it and super intuitive to integrate!

**Comment by tannerlinsley** [comment]
Sure! So let me outline a few quick long-term objectives:
- https://TanStack.com. This is where I want to host the docs for all of my projects (eg. tanstack.com/react-query or react-query.tanstack.com). Right now TanStack.com is just a subscription landing page for my newsletter (and a link to the discord channel).
- Next.js. I really don't want to use anything else on TanStack.com, but I'm also not sure how something like a documentation build would fit into Next, especially if it's sitting at a sub-route or sub-domain for react-query
- Repeatable. I would really like whatever we do with the docs here to also work for the rest of my libraries. Common theming, setup, etc.

That's just the start of what's on my mind though. So to me, I think the first thing we could do is just ship an MVP. We could that using something like docusaurus (v2), host it on netlify, customize the theme a bit to look TanStack-ish, and alias it to react-query.tanstack.com. Then from there we can start to figure out the rest of the outline above.

**Reply by ayush987goyal** [reply]
Hey, you can probably still use Next.js for the documentation instead of docusaurus. Jared Palmer recently created a new docs site for Formik using the similar tech stack and wrote a [blog](https://formik.org/blog/new-docs) about it.

**Reply by tannerlinsley** [reply]
Oh wow. I’m totally on board.

**Reply by MichaelDeBoey** [reply]
@tannerlinsley have you considered using Gatsby for you (docs) site?

**Reply by tannerlinsley** [reply]
Yep. Hard pass

**Comment by otaciliolacerda** [comment]
Let me know if you want help on this.
